File: decl-ts/tests/internal/pipeline_test.ts
// pipeline (tests/internal/checks.json): the source-level report — the
// phase that decided, and what it carries.
import { initParser } from '../../src/node.ts';
import { initFormatter } from '../../src/fmt.ts';
import { evaluateSource } from '../../src/pipeline.ts';
import { check, total } from '../common/check.ts';

await initParser();
await initFormatter();
const parse = evaluateSource('const x = \n');
const checked = evaluateSource('type Bad = 10..3\n');
const clean = evaluateSource('export output x: int = 1\ninput y: int\n');
check(
  'report',
  parse.phase === 'parse' &&
    !parse.ok &&
    parse.parseErrors.length > 0 &&
    checked.phase === 'check' &&
    !checked.ok &&
    checked.checks.some((d) => d.code === 'E4011') &&
    clean.phase === 'evaluate' &&
    clean.ok &&
    JSON.stringify(clean.outputs) === '[{"name":"x","json":"1"}]' &&
    JSON.stringify(clean.inputs) === '["y"]',
  JSON.stringify({ parse, checked, clean }),
);
total();

File: decl-py/tests/internal/pipeline_test.py
"""pipeline (tests/internal/checks.json): the source-level report — the
phase that decided, and what it carries."""

from __future__ import annotations

from decl.pipeline import evaluate_source


def test_report() -> None:
    parse = evaluate_source("const x = \n")
    assert parse["phase"] == "parse" and not parse["ok"] and parse["parse_errors"]
    checked = evaluate_source("type Bad = 10..3\n")
    assert checked["phase"] == "check" and not checked["ok"]
    assert any(d.get("code") == "E4011" for d in checked["checks"])
    clean = evaluate_source("export output x: int = 1\ninput y: int\n")
    assert clean["phase"] == "evaluate" and clean["ok"]
    assert clean["outputs"] == [{"name": "x", "json": "1"}]
    assert clean["inputs"] == ["y"]
